mimic/model/maas_objects.py

```diff
@@ -0,0 +1,229 @@
+"""
+MaaS API data model
+"""
+
+import random
+
+
+def random_letters(len_min, len_max):
+    """
+    Makes a string with random letters (a-z).
+    """
+    return ''.join([chr(ord('a') + random.randint(0, 25))
+                    for _ in xrange(random.randint(len_min, len_max))])
+
+
+class Metric(object):
+    """
+    Models a MaaS metric type.
+    """
+    def __init__(self, name, type, clock, unit='other'):
+        """
+        Initializes the metric.
+        """
+        self.name = name
+        self.type = type
+        self._unit = unit
+        self._overrides = {}
+        self._clock = clock
+
+    def _override_key(self, **kwargs):
+        """
+        Computes a key used for hashing overrides, as a tuple of strings.
+        """
+        return (kwargs['entity_id'],
+                kwargs['check_id'],
+                kwargs.get('monitoring_zone', '__AGENT__'))
+
+    def set_override(self, **kwargs):
+        """
+        Sets the override metric for a given tenant, entity and check.
+
+        Override metrics are defined as a function which takes a timestamp
+        and returns the metric value.
+        """
+        self._overrides[self._override_key(**kwargs)] = kwargs['override_fn']
+
+    def clear_overrides(self):
+        """
+        Clears the override metric values.
+        """
+        self._overrides = {}
+
+    def _get_default_data(self):
```

Comment:
Here's the theory I worked out as I hashed this out a few times. In MaaS, a Metric is a piece of data that represents some specific value, and the value probably changes over time. So, when writing the control API, what you really want to do is to be able to specify a function for a given `(entity, check, metric)` triple that varies over time (or not). It's easy to get from here, to the multiplot mock which returns either random noise or a square wave.

It's also easy to invent metrics to come back with while testing checks without relying on magic numbers; recall the discussion on #275. They're just random, and the user can override the metric results with a custom function (maybe a constant value) that they choose.